xarray/tests/test_netcdftimeindex.py

```diff
@@ -0,0 +1,363 @@
+    ('minute', (2, 2, 10, 6, 2), (2, 2, 10, 6, 2, 59, 999999)),
+    ('second', (2, 2, 10, 6, 2, 8), (2, 2, 10, 6, 2, 8, 999999))
+], ids=['day', 'hour', 'minute', 'second'])
+def test_parsed_string_to_bounds_sub_monthly(date_type, reso,
+                                             ex_start_args, ex_end_args):
+    parsed = date_type(2, 2, 10, 6, 2, 8, 1)
+    expected_start = date_type(*ex_start_args)
+    expected_end = date_type(*ex_end_args)
+
+    result_start, result_end = _parsed_string_to_bounds(
+        date_type, reso, parsed)
+    assert result_start == expected_start
+    assert result_end == expected_end
+
+
+def test_get_loc(date_type, index):
+    result = index.get_loc('0001')
+    expected = [0, 1]
+    assert_array_equal(result, expected)
+
+    result = index.get_loc(date_type(1, 2, 1))
+    expected = 1
+    assert result == expected
+
+    result = index.get_loc('0001-02-01')
+    expected = 1
+    assert result == expected
+
+
+@pytest.mark.parametrize('kind', ['loc', 'getitem'])
+def test_get_slice_bound(date_type, index, kind):
+    result = index.get_slice_bound('0001', 'left', kind)
+    expected = 0
+    assert result == expected
+
+    result = index.get_slice_bound('0001', 'right', kind)
+    expected = 2
+    assert result == expected
+
+    result = index.get_slice_bound(
+        date_type(1, 3, 1), 'left', kind)
+    expected = 2
+    assert result == expected
+
+    result = index.get_slice_bound(
+        date_type(1, 3, 1), 'right', kind)
+    expected = 2
+    assert result == expected
+
+
+def test_date_type_property(date_type, index):
+    assert index.date_type == date_type
+
+
+def test_contains(date_type, index):
+    assert '0001-01-01' in index
+    assert '0001' in index
+    assert '0003' not in index
+    assert date_type(1, 1, 1) in index
+    assert date_type(3, 1, 1) not in index
+
+
+def test_groupby(da):
+    result = da.groupby('time.month').sum('time')
+    expected = xr.DataArray([4, 6], coords=[[1, 2]], dims=['month'])
+    assert_identical(result, expected)
+
+
+@pytest.mark.parametrize('sel_arg', [
+    '0001',
+    slice('0001-01-01', '0001-12-30'),
+    [True, True, False, False]
+], ids=['string', 'string-slice', 'bool-list'])
+def test_sel_string_or_list(da, index, sel_arg):
+    expected = xr.DataArray([1, 2], coords=[index[:2]], dims=['time'])
+    result = da.sel(time=sel_arg)
+    assert_identical(result, expected)
+
+
+def test_sel_date_slice_or_list(da, index, date_type):
+    expected = xr.DataArray([1, 2], coords=[index[:2]], dims=['time'])
+    result = da.sel(time=slice(date_type(1, 1, 1), date_type(1, 12, 30)))
+    assert_identical(result, expected)
+
+    result = da.sel(time=[date_type(1, 1, 1), date_type(1, 2, 1)])
+    assert_identical(result, expected)
+
+
+def test_sel_date_scalar(da, date_type, index):
+    expected = xr.DataArray(1).assign_coords(time=index[0])
+    result = da.sel(time=date_type(1, 1, 1))
+    assert_identical(result, expected)
+
```

Comment:
Add some tests for `sel` (both scalar and list) with both `method='pad'` and `method='nearest'` and optionally setting `tolerance`